moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -246,7 +255,7 @@ class Robot
    */
   bool jointInterpolation(const std::unique_ptr<TrajectoryPoint> & traj_point,
            std::vector<trajectory_msgs::JointTrajectoryPoint> & points,
-           const unsigned int num_steps);
+           const unsigned int num_steps, bool collision_check = false);
```

Comment:
Do you have to add a collision check to multiple methods, or is simpler, cleaner, and easier to maintain if we check in one place.